### PR TITLE
makefile fixes

### PR DESCRIPTION
**Pull request by metux**
Several makefile fixes.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -61,6 +67,7 @@ install:
 	install -D -m 755 $(CURDIR)/_output/nerdctl $(DESTDIR)$(BINDIR)/nerdctl
 	install -D -m 755 $(CURDIR)/extras/rootless/containerd-rootless.sh $(DESTDIR)$(BINDIR)/containerd-rootless.sh
 	install -D -m 755 $(CURDIR)/extras/rootless/containerd-rootless-setuptool.sh $(DESTDIR)$(BINDIR)/containerd-rootless-setuptool.sh
+	install -D -m 644 -t $(DESTDIR)$(DOCDIR)/nerdctl docs/*.md
```
Comment (fahedouch) [inline Makefile:70]:
```suggestion
	install -D -m 644  $(CURDIR)/docs/*.md  $(DESTDIR)$(DOCDIR)/nerdctl
```

File: Makefile [hunk 2]
```diff
@@ -19,19 +19,25 @@
...
+BINDIR  ?= $(PREFIX)/bin
+DATADIR ?= $(PREFIX)/share
+DOCDIR  ?= $(DATADIR)/doc
 
 VERSION ?= $(shell git describe --match 'v[0-9]*' --dirty='.m' --always --tags)
 VERSION_TRIMMED := $(VERSION:v%=%)
 REVISION ?= $(shell git rev-parse HEAD)$(shell if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi)
 
 GO_BUILD_LDFLAGS ?= -s -w
+GO_BUILD_FLAGS ?=
```
[the remark this answers is not in the document]
Comment (metux) [reply Makefile:40]: It's meant to be passed by the caller - just here for reference.

File: Makefile (hunk 1 above)
Comment (metux) [reply Makefile:70]: Nice though, but doesn't work that way: the command won't create the "nerdctl" dir ('-D' creates only the leading path, but explicitly leaving off the last component). That's why I'm using '-t', which does create that dir.

**Comment by metux** [comment]
Looks like the node where "test / test-integration-rootless (20.04, v1.7.3, test-integration-rootless)" is running got a problem - IPFS test gets orphan containers, thus can't remove the image.

**Pull request merged**